### PR TITLE
Shape guard structure

### PR DESCRIPTION
**Pull request by voznesenskym**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/_guards.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import dataclasses
 import enum
 import weakref
-from typing import Callable, List, Optional
+from typing import Callable, List, NamedTuple, Optional
+
+import sympy
```
Comment (voznesenskym) [inline torch/_guards.py:6]: wtf?

Comment (ezyang) [reply torch/_guards.py:6]: I think it's because the lint env isn't installing sympy

**Review by ezyang — APPROVED** [review]

**Comment by voznesenskym** [comment]
Im so tired of linter gaps between local and remote, even after constant lintrunner init

**Pull request closed without merging**